internal/eventstore/eventstore.go

```diff
@@ -0,0 +1,73 @@
+package eventstore
+
+import (
+	"context"
+
+	"github.com/caos/zitadel/internal/errors"
+	"github.com/caos/zitadel/internal/eventstore/internal/repository"
+	"github.com/caos/zitadel/internal/eventstore/models"
+)
+
+type Eventstore interface {
+	AggregateCreator() *models.AggregateCreator
+	Health(ctx context.Context) error
+	PushAggregates(ctx context.Context, aggregates ...*models.Aggregate) error
+	FilterEvents(ctx context.Context, searchQuery *models.SearchQuery) (events []*models.Event, err error)
+}
+
+var _ Eventstore = (*eventstore)(nil)
+
+type eventstore struct {
+	repo             repository.Repository
+	aggregateCreator *models.AggregateCreator
+}
+
+func (es *eventstore) AggregateCreator() *models.AggregateCreator {
+	return es.aggregateCreator
+}
+
+func (es *eventstore) PushEvents(ctx context.Context, aggregates ...[]*models.Event) (err error) {
+	if len(aggregates) == 0 {
+		return errors.ThrowInvalidArgument(nil, "EVENT-JCifQ", "no aggregates provided")
+	}
+
+	for _, events := range aggregates {
+		if len(events) == 0 {
+			return errors.ThrowInvalidArgument(nil, "EVENT-eci1Z", "no events provided")
+		}
+		for _, event := range events {
+			if err := event.Validate(); err != nil {
+				return errors.ThrowInvalidArgument(err, "EVENT-rsa2U", "event invalid")
+			}
+		}
+	}
```

Comment:
this looks wrong for me, you already have events, so you dont have to loop over them and check the length

Reply:
pushEvents is removed